src/lock_free/mwcas_descriptor.cpp

```diff
@@ -208,37 +208,37 @@ MwCASDescriptor::MwCASInternal(  //
   }
 
   const auto succeeded = (cur_stat == kSucceeded);
-  uint64_t ref_cnt{};
+  bool referred = false;
   if (succeeded) {
     for (size_t i = 0; i < target_cnt_; ++i) {
       auto &target = targets_[i];
       const auto ver = (target.old_val.load(kRelaxed) + kVersionUnit) & kVersionMask;
-      ref_cnt += Finalize(base_addr, target, (target.new_val | ver));
+      referred = Finalize(base_addr, target, (target.new_val | ver)) || referred;
     }
   } else {
     for (size_t i = 0; i < target_cnt_; ++i) {
       auto &target = targets_[i];
       const auto val = (target.old_val.load(kRelaxed) + kVersionUnit) & kVerAndValMask;
-      ref_cnt += Finalize(base_addr, target, val);
+      referred = Finalize(base_addr, target, val) || referred;
     }
   }
 
-  return std::pair{succeeded, ref_cnt > 0};
+  return std::pair{succeeded, referred};
 }
 
 auto
 MwCASDescriptor::Finalize(  //
     uint64_t desc_addr,     //
     MwCASTarget &target,    //
     uint64_t desired)       //
-    -> uint64_t
+    -> bool
 {
   auto expected = target.addr->load(kRelaxed);
-  while (((expected ^ desc_addr) & kDescMask) == 0
-         && !target.addr->compare_exchange_weak(expected, desired, kRelaxed, kRelaxed)) {
-    CPP_UTILITY_SPINLOCK_HINT
+  while (true) {
+    if (((expected ^ desc_addr) & kDescMask) != 0) return true;
+    if (target.addr->compare_exchange_weak(expected, desired, kRelaxed, kRelaxed))
+      return (expected & kCntMask) != 0;
```

Comment:
範囲を明示しておきたいので，改行が入るときは常に括弧を入れるようにしておいてください．また，迷いますが，spin lock用のヒント（PAUSE命令）も入れておきましょうか．

```suggestion
    if (target.addr->compare_exchange_weak(expected, desired, kRelaxed, kRelaxed)) {
      return (expected & kCntMask) != 0;
    }
    CPP_UTILITY_SPINLOCK_HINT
```